test/syslogger_SUITE.erl

```diff
@@ -0,0 +1,151 @@
+
+    unload_meck(Config),
+    ok.
+
+init_per_testcase(_TC, Config) ->
+    meck:reset(syslogger),
+    Config.
+
+end_per_testcase(_TC, _Config) ->
+    ok.
+
+%%
+%% Tests
+%%
+syslogger_is_used_to_log(_Config) ->
+    logger:log(error, "Something"),
+
+    ?assert(meck:validate(syslogger)),
+    ok = meck:wait(syslogger, log, [#{level => error,
+                                           msg => {string, "Something"},
+                                           meta => '_'},
+        #{id => ?HANDLER_ID, ident => ?IDENT, facility => ?FACILITY}],
+        timer:seconds(1)).
+
+log_at_every_level(_Config) ->
+    %% given
+    [ begin
+      %% when
+          mongoose_logs:set_global_loglevel(LName),
+          %% then
+          log_at_level(LName)
+      end || {_, LName} <- levels() ].
+
+log_at_level(none) ->
+    %% When log level `none` is set and we log on each possible level...
+    [ logger:log(LevelName, "", []) || {_, LevelName} <- levels(), LevelName /= none ],
+    %% ...then nothing ends up in the log file.
+    %% (polling doesn't make sense in this one case, we have to sleep)
+    Fun = fun() -> meck:history(syslogger) end,
+    async_helper:wait_until(Fun, []);
+log_at_level(LName) ->
+    %% When current log level is L and we log on each possible level...
+    [ logger:log(LevelName, "Syslogger testing") || {_, LevelName} <- levels(), LevelName /= none ],
+    %% ...then for each sensible level (i.e. less or equal to current level)
+    %% we get a call to syslogger.
+    [ok = meck:wait(syslogger, log, [#{level => L,
+                                        msg => {string, "Syslogger testing"},
+                                        meta => '_'}, '_'], 100)
+     || L <- levels_less_than_or_equal_to(LName)].
+
+log_at_custom_level(_Config) ->
+    %% given logging on custom log level for the helper module
+    mongoose_logs:set_global_loglevel(critical),
+    mongoose_logs:set_module_loglevel(mongooseim_loglevel_SUITE_helper, debug),
+    %% when we log from here and from the helper module
+    NotSupposedToBeLogged = "This should not be in the logs!",
+    ShouldBeSuccessfullyLogged = "This MUST appear in the logs successfully",
+    logger:log(debug, NotSupposedToBeLogged),
+    mongooseim_loglevel_SUITE_helper:log(debug, ShouldBeSuccessfullyLogged),
+    ok = meck:wait(syslogger, log, [#{level => debug,
+                                      msg => {string, ShouldBeSuccessfullyLogged},
+                                      meta => '_'}, '_'], 100),
+
+    Fun = fun() -> meck:num_calls(syslogger, log, '_') end,
+    async_helper:wait_until(Fun, 1).
+
+%%
+%% Helpers
+%%
+
+setup_meck(_Config) ->
+    meck:new(syslogger, [no_link]),
+    meck:expect(syslogger, adding_handler,
+                fun(C) ->
+                    {ok, C}
+                end),
+    meck:expect(syslogger, log,
+                fun(_LogEvent, _Config) ->
+                    ok
+                end).
+
+levels() ->
```

Comment:
I'm not sure this is needed to be copied from the other test suite, but if so, perhaps it could be just moved to the helper `mongoose_loglevel_SUITE_helper` module and used once from both this test suite and the other one, together with anything else that seems sensible to be moved around to avoid code duplication.

Reply:
I wasn't sure either, whether it was at all needed here, but I copied it as I followed the example of the `mongoose_loglevel_SUITE` when writing. I will move the common parts to the helper.